infra/helper.py

```diff
@@ -162,15 +162,12 @@ def main():
   args = parser.parse_args()
 
   # We have different default values for `sanitizer` depending on the `engine`.
-  try:
-    if not args.sanitizer:
-      if args.engine == 'dataflow':
-        args.sanitizer = 'dataflow'
-      else:
-        args.sanitizer = 'address'
-  except:
-    # Some commands do not have `sanitizer` argument.
-    pass
+  # Some commands do not have `sanitizer` argument, so `hasattr` is necessary.
+  if hasattr(args, 'sanitizer') and not args.sanitizer:
```

Comment:
nit: Sorry should've suggested `if not getattr(args, 'sanitizer', None)` as it is shorter.

Reply:
If I lose this disagreement over whether a seperate engine exists then please fail loudly if --sanitizer dataflow is specified but engine isn't data flow.

Reply:
I think this won't be that useful, as I need to distinguish whether the argument is not used at all (the first condition) or is not specified (the second condition) and needs a default value.

Reply:
> If I lose this disagreement over whether a seperate engine exists then please fail loudly if --sanitizer dataflow is specified but engine isn't data flow.

Done, now re-building images and testing locally.

Reply:
> I think this won't be that useful, as I need to distinguish whether the argument is not used at all (the first condition) or is not specified (the second condition) and needs a default value.

Ah right my bad.

Reply:
> > If I lose this disagreement over whether a seperate engine exists then please fail loudly if --sanitizer dataflow is specified but engine isn't data flow.
> 
> Done, now re-building images and testing locally.

Thanks!